server/src/main/java/org/elasticsearch/cluster/metadata/MetadataUpdateSettingsService.java

```diff
@@ -304,8 +311,10 @@ private static void maybeUpdateClusterBlock(String[] actualIndices, ClusterBlock
                 } else {
                     blocks.removeIndexBlock(index, block);
                 }
+                changed = true;
```

Comment:
This sets `changed=true` regardless of whether block was in place or not. I think we should handle that too - mainly for completeness more than necessity.

Reply:
I was focusing on the optimization for CCR. I've adjusted this in https://github.com/elastic/elasticsearch/pull/61348/commits/f43c2ead444482eebbd402773e847eb1a9e086b6.